azerothcore/azerothcore-wotlk
# fix(DB/loot): Prospecting Thorium doesn't always yield a gem

## Changes Proposed:
-  Fix the grouping of thorium ore prospecting loot so you are guaranteed to get either a tbc gem or a classic gem

## Issues Addressed:

- Closes #5781
- Closes chromiecraft/chromiecraft#600

## SOURCE:


## Tests Performed:

- Ran ```SET @THORIUM_ORE = 10620;
SELECT * FROM `acore_world`.`prospecting_loot_template` WHERE ENTRY = @THORIUM_ORE``` and checked the results to ensure the GroupIds were now correct.
- Prospected 100 Thorium ore after the update and never got an empty result.


## How to Test the Changes:


1. Before the update: Prospect Thorium Ore (https://wowgaming.altervista.org/aowow/?item=10620#prospecting).This will sometimes give you nothing.
2. After the update this should always give you either a TBC gem or a classic gem.
3.  Check the GroupIds using ```SET @THORIUM_ORE = 10620;
SELECT * FROM `acore_world`.`prospecting_loot_template` WHERE ENTRY = @THORIUM_ORE```

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
